pkg/render/node.go

```diff
@@ -491,11 +491,12 @@ func (c *nodeComponent) nodeDaemonset(cniCfgMap *v1.ConfigMap) *apps.DaemonSet {
 
 // nodeTolerations creates the node's tolerations.
 func (c *nodeComponent) nodeTolerations() []v1.Toleration {
-	return []v1.Toleration{
-		{Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoSchedule},
```

Comment:
Are we sure it's OK to remove these tolerations?

Reply:
I don't have any evidence for efficacy (docs or otherwise). What do they do?

Reply:
These tolerate any taint which has the effect "NoSchedule" or "NoExecute", no matter what the key / value for the particular taint is. They act as a "catch-all".

Reply:
> These tolerate any taint which has the effect "NoSchedule" or "NoExecute", no matter what the key / value for the particular taint is. They act as a "catch-all".

K8s does this for empty keys/effect with operator of exists:
From here: https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
"Note: There are two special cases: An empty key with operator Exists matches all keys, values and effects which means this will tolerate everything. An empty effect matches all effects with key key."

But, I haven't read anything that states you can use an exists operator with a stated effect and an empty key to get wildcard behavior...but perhaps what you are describing is undocumented...or I'm not understanding properly! 

However, the evidence that I have suggests that these are not working as intended. In the DockerEE case, the nodes have a "not-ready" taint with effect of NoSchedule and our calico-node pods would not get deployed.

Reply:
Yeah, it's possible that the behavior has changed. That documentation isn't super clear about it, but does imply that you must leave both key and effect empty in order to get the "match all" behavior.

However, I do know that we have used these particular tolerations on our DaemonSets in order to get this behavior for a very long time. Here's a v3.0 manifest that has this toleration on it! https://github.com/projectcalico/calico/blob/release-legacy/v3.0/getting-started/kubernetes/installation/hosted/calico.yaml#L112-L114

Maybe newer versions of k8s have tightened up what they allow? I'd expect API validation to catch an invalid configuration, though definitely not impossible that k8s has a bug here. 

> However, the evidence that I have suggests that these are not working as intended. In the DockerEE case, the nodes have a "not-ready" taint with effect of NoSchedule and our calico-node pods would not get deployed. With a "not-ready" toleration added, calico-node gets deployed.

What version of k8s is that DockerEE cluster? I'm interested in seeing if I can chase this down.